witchcraft-server/src/debug/diagnostic_types.rs

```diff
@@ -48,6 +45,12 @@ impl Diagnostic for DiagnosticTypesDiagnostic {
     }
 
     fn result(&self) -> Result<Bytes, Error> {
-        Ok(self.types.clone())
+        let mut types: Vec<String> = Vec::new();
+        types.push(DIAGNOSTIC_TYPES_V1.to_string());
+        if let Some(registry) = self.registry.upgrade() {
+            types.push(registry.diagnostics.read().keys().cloned().collect());
```

Comment:
I think you're going to want to do `types.extend(registry.diagnostics.lock().keys().cloned())` - right now you're concatenating all of the names into one big string.

Reply:
🤦